packages/wrangler/src/proxy.ts

```diff
@@ -651,6 +651,11 @@ export async function waitForPortToBeAvailable(
 		}
 
 		function checkPort() {
+			if (port === 0) {
```

Comment:
...oh okay

Reply:
Yup. I'll tidy this up in a subsequent PR removing these race-y functions.